engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/NadelDeepRenameTransform.kt

```diff
@@ -305,31 +302,55 @@ internal class NadelDeepRenameTransform : NadelTransform<NadelDeepRenameTransfor
         state: State,
     ): List<NadelResultInstruction> {
         val parentNodes = getNodesAt(
-                result.data,
-                field.listOfResultKeys.dropLast(1),
-                flatten = true,
+            result.data,
+            field.listOfResultKeys.dropLast(1),
+            flatten = true,
         )
 
         return parentNodes.flatMap { parentNode ->
             @Suppress("UNCHECKED_CAST") // Ensure the result is a Map, return if null
-            val parentMap = parentNode.value as JsonMap? ?: return@flatMap emptyList()
-            val deepRenameInstruction = getMatchingDeepRename(parentMap, state, executionPlan)
+            val parentMap = parentNode.value as JsonMap?
+                ?: return@flatMap makeRemoveFieldInstructions(state, parentNode)
+            val deepRenameInstruction = getMatchingDeepRenameInstruction(parentMap, state, executionPlan)
+                ?: return@flatMap makeRemoveFieldInstructions(state, parentNode)
 
             val nodeToMove = getNodesAt(parentNode, getPathOfNodeToMove(state, deepRenameInstruction))
-                    .emptyOrSingle() ?: return@flatMap emptyList()
+                .emptyOrSingle() ?: return@flatMap emptyList()
 
             listOf(
                 NadelResultInstruction.Copy(
                     subjectPath = nodeToMove.path,
                     destinationPath = parentNode.path + field.resultKey,
                 ),
+            ) + makeRemoveFieldInstructions(state, parentNode, deepRenameInstruction)
+        }
+    }
+
+    /**
+     * Cleans up the [parentNode] by removing fields that were added by this transform.
+     */
+    private fun makeRemoveFieldInstructions(
```

Comment:
Extract these into a function so that we can return these for every node, so that we always clean up the artificial fields we add.

In the future these should be auto removed, but for now just did this.